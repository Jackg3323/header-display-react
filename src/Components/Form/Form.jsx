export default function Form() {
  return (
    <form>
      <div>
        <label htmlFor="name" className="sr-only">
          Name
        </label>
        <input type="text" placeholder="Enter Your Name" id="name" />
        <button type="submit">Submit</button>
      </div>
    </form>
  );
}
