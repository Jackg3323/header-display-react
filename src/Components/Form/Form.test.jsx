import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Form from "./Form";

it("renders without crashing", () => {
  const rendered = renderer.create(<Form setName={() => {}} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("calls the submit handler when the form is submitted", () => {
  const handleSubmit = jest.fn();
  render(<Form setName={handleSubmit} />);
  const submitBtn = screen.getByRole("button", { name: /submit/i });
  userEvent.click(submitBtn);
  expect(handleSubmit).toHaveBeenCalled();
});
