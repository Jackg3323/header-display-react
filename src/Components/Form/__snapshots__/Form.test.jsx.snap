// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<form
  className="mx-auto w-80 flex-col items-center"
  onSubmit={[Function]}
>
  <div>
    <label
      className="sr-only"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="rounded border pl-2"
      id="name"
      name="name"
      placeholder="Enter Your Name"
      type="text"
    />
  </div>
  <button
    className="mt-4 bg-purple-500 px-4 py-2 text-white"
    type="submit"
  >
    Submit
  </button>
</form>
`;
